server/src/test/java/io/crate/execution/dml/upsert/SourceFromCellsTest.java

```diff
@@ -427,4 +427,46 @@ public void test_default_clauses_are_evaluated_per_row() throws Exception {
         assertThat(id1, Matchers.notNullValue());
         assertThat(id1, Matchers.not(is(id2)));
     }
+
+    @Test
+    public void test_generated_partition_column_passes_validation_if_value_matches() throws Exception {
+        var e = SQLExecutor.builder(clusterService)
+            .addPartitionedTable("create table tbl (ts timestamp with time zone, g_ts_month as date_trunc('month', ts)) partitioned by (g_ts_month)")
+            .build();
+        Reference ts = (Reference) e.asSymbol("tbl.ts");
+        Reference g_ts_month = (Reference) e.asSymbol("tbl.g_ts_month");
+        DocTableInfo tbl = e.resolveTableInfo("tbl");
+        InsertSourceGen sourceGen = InsertSourceGen.of(
+            txnCtx,
+            e.nodeCtx,
+            tbl,
+            new PartitionName(tbl.ident(), List.of("1630454400000")).asIndexName(),
+            GeneratedColumns.Validation.VALUE_MATCH,
+            List.of(ts, g_ts_month)
+        );
+        Map<String, Object> generateSourceAndCheckConstraints = sourceGen.generateSourceAndCheckConstraints(new Object[] { 1631628823105L, 1630454400000L });
```

Comment:
pretty minor, how about? 
```suggestion
        Map<String, Object> generateSourceAndCheckConstraints = sourceGen.generateSourceAndCheckConstraints(new Object[] { 1631628823105L, 1630454400000L });
        assertTrue(sourceGen instanceof InsertSourceFromCells);
```

Reply:
Pushed some minor formatting changes